Documentation/platforms/arm/stm32f3/index.rst

```diff
@@ -215,6 +215,13 @@ There are two version of the FPU support built into the STM32 port.
 
      CONFIG_ARCH_FPU=y
 
+Flashing
+========
+
+NuttX firmware flashing with STLink probe and OpenOCD::
+
+    openocd -f  interface/stlink.cfg -f target/stm32f3x.cfg -c 'program nuttx.bin 0x08000000'
```

Comment:
Might be worth changing last part to 

```program nuttx.bin 0x08000000; reset run; exit```

This will also reset chip and exit from openocd - otherwise openocd will be running and waiting for GDB input.

Also it may be useful to add simply reset command as well

```openocd -f  interface/stlink.cfg -f target/stm32f3x.cfg -c 'init; reset run; exit'```

Reply:
btw, these should also work for f4/f7 and h7. Default flash start for these chips should be the same (`0x08000000`). All you need to change is target, stm32f4x/stm32f7x etc. You can find these in `/usr/share/openocd/scripts`

Reply:
Thank you @mlyszczek !! Can you also provide simple instruction on how to debug? We will bundle all info in Flashing and Debugging section :-)

Reply:
Only partially since I do not use any GUI tools. Generally if you skip `exit` in any command, openocd will stay open with gdbserver running and waiting for gdb to connect.

With gdb you can connect to server with command

```arm-none-eabi-gdb nuttx```

add `--tui` for some better code visuals

```arm-none-eabi-gdb --tui nuttx```

After starting gdb you must connect to the server, inside gdb run

```target remote :3333```

Or use ```-ex``` gdb command to connect automaticall

```arm-none-eabi-gdb --tui nuttx -ex "target remote 127.0.0.1:3333"```

Now mind you, I'm 99% these will work, but I took these from my shell history as I do not have any stm connected right now. I'm pretty sure port `3333` is default for openocd.

Inside gdb you can restart flow with `mon reset halt` or `mon reset run` depending on what you want. Then you can use standard gdb commands. `ctr+c` to pause execution where you can add breakpoints. `b <function-name>|<file-name:line-number>` to add breakpoint. `c` to continue until next breakpoint is hit. `n` to jump to next line (without going into function). `s` to step into the function. `u` to run until loop is done. `bt` to see current stacktrace when execution is paused. `p <var-name>` to print value of variable and so on, more in the gdb manual. Also it's important to at least enable debuging symbols. If code is built with `-O2` code might jump in unexpected ways and some variables might be optimized out and not available to print. Using `-O0` will fix those issues.

I don't know how the gui debugging works, but I suppose they should be able to simply connect to local port `3333` and start debugging. 

Do with these what you want. Redact it, paste it as is. If you have more questions just ping me.

Reply:
Thank you :-) Updated STM32F[3,4,7] please verify :-)